File: backend/src/services/emailNotificationService.js
const nodemailer = require('nodemailer');

class EmailNotificationService {
  constructor() {
    this.mailEnabled = String(process.env.MAIL_ENABLED || 'false').toLowerCase() === 'true';
    this.fromEmail = process.env.MAIL_FROM || 'no-reply@complementa.local';
  }

  async enviarStatusSolicitacao(destino, tituloAtividade, status, horasAprovadas, justificativa) {
    if (!destino || !String(destino).trim()) return;

    const assunto = '[Complementa+] Atualizacao da sua solicitacao';
    const corpo = this.montarCorpo(tituloAtividade, status, horasAprovadas, justificativa);

    if (!this.mailEnabled) {
      console.log('[mail disabled]', { destino, assunto, corpo });
      return;
    }

    // Config padrão: sem SMTP definido, falhar iria afetar; por segurança, mantemos desabilitado por padrão.
    if (!process.env.SMTP_HOST) {
      console.log('[mail enabled but SMTP not configured] skipping');
      return;
    }

    const transporter = nodemailer.createTransport({
      host: process.env.SMTP_HOST,
      port: Number(process.env.SMTP_PORT || 587),
      secure: false,
      auth: process.env.SMTP_USER
        ? { user: process.env.SMTP_USER, pass: process.env.SMTP_PASS }
        : undefined
    });

    await transporter.sendMail({
      from: this.fromEmail,
      to: String(destino).trim().toLowerCase(),
      subject: assunto,
      text: corpo
    });
  }

  montarCorpo(tituloAtividade, status, horasAprovadas, justificativa) {
    let texto = '';
    texto += 'Sua solicitacao foi atualizada no Complementa+.\n\n';
    texto += `Atividade: ${tituloAtividade == null ? '-' : tituloAtividade}\n`;
    texto += `Status: ${status}\n`;

    if (status === 'APROVADO' && horasAprovadas != null) {
      texto += `Horas aprovadas: ${horasAprovadas}\n`;
    }
    if (status === 'REPROVADO' && justificativa != null && String(justificativa).trim().length) {
      texto += `Motivo da recusa: ${String(justificativa).trim()}\n`;
    }

    texto += '\nAcesse o sistema para mais detalhes.';
    return texto;
  }
}

module.exports = EmailNotificationService;

File: backend/app/services/email_notification_service.rb
# frozen_string_literal: true

class EmailNotificationService
  def enviar_status_solicitacao(destino, titulo_atividade, status, horas_aprovadas: nil, justificativa: nil)
    return if destino.blank?

    assunto = "[Complementa+] Atualizacao da sua solicitacao"
    corpo = montar_corpo(titulo_atividade, status, horas_aprovadas, justificativa)
    Rails.logger.info("Envio de email desabilitado. Destino=#{destino} Assunto=#{assunto} Corpo=#{corpo}")
  end

  private

  def montar_corpo(titulo_atividade, status, horas_aprovadas, justificativa)
    texto = +"Sua solicitacao foi atualizada no Complementa+.\n\n"
    texto << "Atividade: #{titulo_atividade.presence || '-'}\n"
    texto << "Status: #{status}\n"

    if status == "APROVADO" && horas_aprovadas
      texto << "Horas aprovadas: #{horas_aprovadas}\n"
    end

    if status == "REPROVADO" && justificativa.present?
      texto << "Motivo da recusa: #{justificativa.strip}\n"
    end

    texto << "\nAcesse o sistema para mais detalhes."
    texto
  end
end
